bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/VoiceManagerImpl.java

```diff
@@ -736,11 +753,47 @@ public Collection<KSService> getKSs() {
         return hli;
     }
 
+    @Override
+    public @Nullable Collection<HumanLanguageInterpreter> getDefaultHLIs() {
+        var defaultHLI = this.defaultHLI;
+        if (defaultHLI != null) {
+            var interpreters = this.getHLIByIds(defaultHLI);
+            if (interpreters != null && !interpreters.isEmpty()) {
+                return interpreters;
+            }
+        }
+        if (!humanLanguageInterpreters.isEmpty()) {
+            var hli = humanLanguageInterpreters.values().iterator().next();
+            if (hli != null) {
+                return List.of(hli);
+            }
+        }
+        logger.debug("No HumanLanguageInterpreter available!");
+        return null;
+    }
+
     @Override
     public @Nullable HumanLanguageInterpreter getHLI(String id) {
         return humanLanguageInterpreters.get(id);
     }
 
+    @Override
+    public @Nullable Collection<HumanLanguageInterpreter> getHLIByIds(String idList) {
+        List<HumanLanguageInterpreter> interpreters = new ArrayList<>();
+        for (var id : idList.split(",")) {
+            var hli = humanLanguageInterpreters.get(id);
+            if (hli == null) {
+                logger.warn("Default HumanLanguageInterpreter '{}' not available!", id);
+            } else {
+                interpreters.add(hli);
+            }
+        }
+        if (interpreters.isEmpty()) {
```

Comment:
Shouldn't we return null as soon as one of the interpreter is not found (rather than ignoring it) ?

Reply:
I also have doubts about it.
As I was thinking on support configuring a chain of interpreters as default I though it makes more sense not to fail, to prevent errors in case one of the configured interpreters is uninstalled.

Reply:
Ok but in this case, please mention it in the method description (in interface).